src/utils/Configuration.ts

```diff
@@ -0,0 +1,100 @@
+// @ts-ignore
+import * as yml from "node-yaml";
+import {IInvokeConfig, IMOTConfig, IS3Config} from "../models";
+
+/**
+ * Configuration class for retrieving project config
+ */
+class Configuration {
+
+    private static instance: Configuration;
+    private readonly config: any;
+
+    private constructor(configPath: string, secretsPath: string) {
+        const config = yml.readSync(configPath);
+        const secrets = yml.readSync(secretsPath);
+        config.mot.api_key = secrets.mot.api_key;
+
+        // Replace environment variable references
+        let stringifiedConfig: string = JSON.stringify(config);
+        const envRegex: RegExp = /\${(\w+\b):?(\w+\b)?}/g;
+        const matches: RegExpMatchArray | null = stringifiedConfig.match(envRegex);
+
+        if (matches) {
+            matches.forEach((match: string) => {
+                envRegex.lastIndex = 0;
+                const captureGroups: RegExpExecArray = envRegex.exec(match) as RegExpExecArray;
+
+                // Insert the environment variable if available. If not, insert placeholder. If no placeholder, leave it as is.
+                stringifiedConfig = stringifiedConfig.replace(match, (process.env[captureGroups[1]] || captureGroups[2] || captureGroups[1]));
+            });
+        }
+
+        this.config = JSON.parse(stringifiedConfig);
+    }
+
+    /**
+     * Retrieves the singleton instance of Configuration
+     * @returns Configuration
+     */
+    public static getInstance(): Configuration {
+        if (!this.instance) {
+            this.instance = new Configuration("../config/config.yml", "../config/secrets.yml");
+        }
+
+        return Configuration.instance;
+    }
+
+    /**
+     * Retrieves the entire config as an object
+     * @returns any
+     */
+    public getConfig(): any {
+        return this.config;
+    }
+
+    /**
+     * Retrieves the Lambda Invoke config
+     * @returns IInvokeConfig
+     */
+    public getInvokeConfig(): IInvokeConfig {
+        if (!this.config.invoke) {
+            throw new Error("Lambda Invoke config is not defined in the config file.");
+        }
+
+        // Not defining BRANCH will default to local
+        const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
+
+        return this.config.invoke[env];
+    }
+
+    /**
+     * Retrieves the S3 config
+     * @returns IS3Config
+     */
+    public getS3Config(): IS3Config {
+        if (!this.config.s3) {
+            throw new Error("DynamoDB config is not defined in the config file.");
+        }
+
+        // Not defining BRANCH will default to local
+        const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
+
+        return this.config.s3[env];
+    }
+
+    /**
+     * Retrieves the MOT config
+     * @returns IMOTConfig
+     */
+    public getMOTConfig(): IMOTConfig {
```

Comment:
dont think we remotely are doing anything with MOT here. Thats only used by Cert Gen service

Reply:
we are storing apiKey for gov notify service in MOT config

Reply:
exactly my point. Its an API Key for Gov Notify (not MOT) to differentiate environments. API Key you use in CERT GEN for MOT is actually an API key for MOT Certificate Generation service. Variable name should be accordingly. This service has nothing to do with MOT certificate generation service. Any confusion, give me a call pls